internal/controller/cluster_controller.go

```diff
@@ -254,12 +263,127 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	// We need to requeue to check the teleport token validity
 	// and update secret for the cluster, if it expires
+	log.Info("Reconcile completed", "duration", time.Since(start))
 	return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&capi.Cluster{}).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.findClustersForKubeconfigSecret),
+			builder.WithPredicates(predicate.NewPredicateFuncs(r.isKubeconfigSecret)),
+		).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(r.findClustersForTbotConfigMap),
+			builder.WithPredicates(predicate.NewPredicateFuncs(r.isTbotConfigMap)),
+		).
 		Complete(r)
 }
+
+// isKubeconfigSecret checks if a secret is a teleport kubeconfig secret we should watch
+func (r *ClusterReconciler) isKubeconfigSecret(obj client.Object) bool {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		return false
+	}
+
+	// Check if it's in the teleport bot namespace and matches our naming pattern
+	if secret.Namespace != key.TeleportBotNamespace {
+		return false
+	}
+
+	// Check if it matches teleport kubeconfig secret naming pattern: teleport-{cluster}-kubeconfig
+	return strings.HasPrefix(secret.Name, "teleport-") && strings.HasSuffix(secret.Name, "-kubeconfig")
+}
+
+// isTbotConfigMap checks if a configmap is a tbot configmap we should watch
+func (r *ClusterReconciler) isTbotConfigMap(obj client.Object) bool {
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return false
+	}
+
+	// Check if it's in the teleport bot namespace and matches our naming pattern
+	if cm.Namespace != key.TeleportBotNamespace {
+		return false
+	}
+
+	// Check if it matches tbot configmap naming pattern: teleport-tbot-{cluster}-config
+	return strings.HasPrefix(cm.Name, "teleport-tbot-") && strings.HasSuffix(cm.Name, "-config")
+}
```

Comment:
If these are truly necessary, they should be `key` functions, not instance methods